wp-includes/rest-api/endpoints/class-wp-rest-customize-sections-controller.php

```diff
@@ -0,0 +1,374 @@
+<?php
+/**
+ * REST API: WP_REST_Customize_Sections_Controller class
+ *
+ * @package WordPress
+ * @subpackage REST_API
+ * @since ?.?.?
+ */
+
+/**
+ * Core class to access sections via the REST API.
+ *
+ * @since ?.?.?
+ *
+ * @see WP_REST_Controller
+ */
+class WP_REST_Customize_Sections_Controller extends WP_REST_Controller {
+
+	/**
+	 * Array of sections.
+	 *
+	 * @since 4.?.?
+	 * @access protected
+	 * @var array
+	 */
+	protected $sections = array();
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 4.7.0
+	 * @access public
+	 */
+	public function __construct() {
+		$this->namespace = 'customize/v1';
+		$this->rest_base = 'sections';
+
+		if ( ! class_exists( 'WP_Customize_Manager' ) ) {
+			require_once( ABSPATH . WPINC . '/class-wp-customize-manager.php' );
+		}
+	}
+
+	/**
+	 * Ensure customize manager.
+	 *
+	 * @return WP_Customize_Manager Manager.
+	 * @global WP_Customize_Manager $wp_customize
+	 */
+	public function ensure_customize_manager() {
+		global $wp_customize;
+		if ( empty( $wp_customize ) ) {
+			$wp_customize = new WP_Customize_Manager(); // WPCS: global override ok.
+		}
+
+		if ( ! did_action( 'customize_register' ) ) {
+
+			/** This action is documented in wp-includes/class-wp-customize-manager.php */
+			do_action( 'customize_register', $wp_customize );
+		}
+
+		if ( empty( $this->sections ) ) {
+
+			$wp_customize->prepare_controls();
+
+			// Get all the sections like this since some sections won't be directly accessible after prepare_controls().
+			$this->sections = array_merge( $wp_customize->sections() );
+			foreach ( $wp_customize->panels() as $panel ) {
+				$this->sections = array_merge( $this->sections, $panel->sections );
+			}
```

Comment:
Yes, it's because unit tests currently don't clean up after the creation of the `$wp_customize` global in each test. I've done so in 08a0845 and running all tests pass for me locally now.